docs/swagger.yaml

```diff
@@ -180,6 +180,134 @@ paths:
+                  $ref: '#/components/schemas/MerchantCallbackQr'
+        '400':
+          $ref: '#/components/responses/400BadRequest'
+        '401':
+          description: Unauthorized
+        '500':
+          description: Internal Server Error
+      description: Returns all QR codes that matches the provided Merchant-Serial-Number.
+      parameters:
+        - $ref: '#/components/parameters/Authorization'
+        - $ref: '#/components/parameters/Ocp-Apim-Subscription-Key'
+        - $ref: '#/components/parameters/Merchant-Serial-Number-Required'
+      tags:
+        - Merchant callback QR
+  '/v1/merchant-callback/{merchantQrId}':
+    get:
+      summary: Get callback QR by id
+      operationId: GetMerchantCallbackById
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MerchantCallbackQr'
+        '400':
+          $ref: '#/components/responses/400BadRequest'
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+        '500':
+          description: Internal Server Error
+      description: Returns the QR code represented by the merchantQrId and Merchant-Serial-Number provided in the path and header respectively.
+      parameters:
+        - $ref: '#/components/parameters/Authorization'
+        - $ref: '#/components/parameters/Ocp-Apim-Subscription-Key'
+        - $ref: '#/components/parameters/Merchant-Serial-Number-Required'
+      tags:
+        - Merchant callback QR
+    put:
+      summary: Create or update callback QR
+      operationId: PutMerchantCallbackQr
+      description: |-
+        Creates or updates the QR code that encapsulates the provided merchantSerialNumber and merchantQrId. See [link to webhooks API] to create a webhook that will send callbacks when this QR code is scanned by a Vipps user.
+
+        If the endpoint is called with the same merchantQrId twice or more, it is the last call that defines the imageFormat and size (both provided in the headers) and location. The actual QR code image will not be updated on consecutive calls.
+      parameters:
+        - $ref: '#/components/parameters/Accept'
+        - $ref: '#/components/parameters/Size'
+        - $ref: '#/components/parameters/Authorization'
+        - $ref: '#/components/parameters/Ocp-Apim-Subscription-Key'
+        - $ref: '#/components/parameters/Merchant-Serial-Number-Required'
+      responses:
+        '200':
+          description: OK
+        '400':
+          $ref: '#/components/responses/400BadRequest'
+        '401':
+          description: Unauthorized
+        '415':
+          $ref: '#/components/responses/415UnsupportedMediaType'
+        '500':
+          description: Internal Server Error
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/MerchantCallbackPutRequest'
```

Comment:
I would argue that we should return the #/components/schemas/MerchantCallbackQr here as well, there is no reason not to do it

And hmm, i kind of like the idea of put, but we need to know exactly how we should handle the merchant doing a put on the same id. What should happen then?

Reply:
put should ideally change the whole object, not only parts of it. But, in this case the unique id for the QR (our internal hex) should be the same, right? 
I prefer post + delete to ensure that merchants doesnt do anything stupid by mistake (e.g. if they accidentaly do an api call with an existing id)

Reply:
Also, I don't think we should return the QR code in the response. The reason is, that the endpoints should be very specific. The PUT endpoint is just for creating/updating, but if they need to retrieve the resource, there is a specific endpoint for that (the GET endpoints). Some of the benefits with splitting it up, is that we are absolutely sure that merchants that call the PUT endpoint they clearly intent to update it. If we return the QR code, we would not know if the merchants call the PUT endpoint to update, or just to get the QR code. If we return the QR Code there would not be a reason to have the GET endpoint actually.
Another reason is related to async. If we suddenly are not able to generate a QR and return the URL before the request timeout. Its better to be eventual consistent, so they call the GET endpoint and then they might have to retry the request until the resource is ready

Reply:
The internal HEX is not part of the API. Its not mentioned anywhere. So what constitutes the ressource in this API is the object that include MSN, MerchantQrId, size, format and location. When they call the PUT endpoint they will replace all of this (except the MSN and MerchantQrId which combined is the id for the resource). So in that regard it is not a patch. In regards of merchants doing something stupid by mistake, that can always happen :) - The best prevention of client errors is an API that is concise, and easy to understand

Reply:
When it comes to deciding between using `PUT` or `POST` in our API, I  believe that `POST` is the more intuitive and natural choice.  I think that using `POST` just feels more straightforward and easier to understand. 

If I were to implement an API which _creates_ a resource with a `PUT` request, I would double-check that I had selected the correct endpoint.

Reply:
I agree that intuitively POST is what you look for when you create a resource. But when there is no POST endpoint, but only a PUT, wouldn't that confusion be quickly resolved you think? - The important part for me for choosing the PUT endpoint is that the resource id is known pre-hand. That makes the action idempotent since no new id is created each time you call the endpoint. So if we make it as a POST endpoint, then the confusion would be looking for a PUT endpoint when you need to 'update' the resource. Also a POST endpoint is not expected to be idempotent, but a PUT is. So all in all, I would argue that the PUT endpoint is most suitable

Reply:
Based on MDN Docs PUT seems to be the correct way to go here
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT

Reply:
As for the response, I think empty response seems most inline with the API RFC
https://github.com/vippsas/rfc/blob/master/docs/0001-api-guidelines.md#create-or-replace-a-thing

Reply:
I feel its really counter intuitive to:

1. create a QR
2. get the QR from the merchant api
3. get the QR image

it seem way overkill to have to do 3 api calls to get the QR image of one QR

Reply:
Well, I agree that the best design would be one GET endpoint that returned the actual QR image. But since we cannot do that, we at least have two actions. One for creating and one for getting the actual image. I guess the big question is if one extra API call is worth the extra benefits. I see the following benefits:
- Each endpoint has a clear intent. Its easy to see from logs what the merchants intent is (do they want to get the QR or update some info on the existing)
- More RESTful.
- Asynchronous API
- Future modifiability. If we suddenly want to change the response, its only the GET endpoint we need to change. Or one less endpoint we need to deprecate.
- Follows our API RFC
That is the arguments I can come up with. How big of a difference do you think it is for the merchants to call one extra endpoint? - They already have all the security (tokens etc), and headers setup. So they only need to add a call to GET with the setup they have already implemented.

Reply:
Why are there two get endpoints?

Reply:
the qr-api only returns a link to the image, not the image itself (so that unauthenticated POS screens can fetch this image directly)
so the flow here would be to create a resource, get the resource from the api (which contains a link to the image), and then get the image. I would argue the middle step is not necessary

Reply:
> Why are there two get endpoints?

To make it possible to either get a specific QR code with a merchantQrId. Or get all QR codes created under the same sale unit. I guess the last one is useful to get all the QR codes for a store if they want to get them all at once for printing purposes for example

Reply:
Well, i'm not sure it's more restful or follows our RFC. The RFC states that a put should return the reference for getting the resource, and in this case it feels like the refernce that can be get'ed would be the image and not the QR. We would still have a get on spesific id's, so we'd still known what merchants are trying to do. 
But, it really doesnt matter that much, its not an api that will be used frequently, only when they create new QRs. Thomas can get the pleasure of deciding 

<img width="985" alt="image" src="https://user-images.githubusercontent.com/47554984/221576498-560220ea-6950-4060-9597-f2684661f2d8.png">

Reply:
We decided to keep the PUT endpoint. Discussed internally on meeting.